McMasterNEUDOSE/cFS
# Adding LibCSP to CFS build

**Describe the contribution**
- Added the libcsp source code to `apps/libcsp`
- Created a script `run.sh` to build CSP and CFS
- CSP is built as a shared library in `apps/libcsp/build`
- The shared library is then copied (installed) to the cf build `build/exe/cpu1/cf/libcsp.so`
- The CSP functions can be included in source code via `#include <csp/csp.h>`

**Testing performed**
1. Added `#include <csp/csp.h>` in `to_lab_app.c`
1. Added a call to `csp_init()` in `TO_LAB_init()` and checked for errors
1. Run `chmod +x run.sh` to make script executable
1. Run `./run.sh` to execute the script
1. Observe build output for errors - **none found**
1. Observed CFS log for errors - **none found**

**Expected behavior changes**
 - API Change: New API for CSP functions

**System(s) tested on**
 - OS: Ubuntu 18.04

**Additional context**
- Next steps will be incorporating the functionality in the CI/TO apps

**Contributor Info - All information REQUIRED for consideration of pull request**
Stephen Scott, McMaster NEUDOSE
